components/server/src/user/user-controller.ts

```diff
@@ -377,8 +409,21 @@ export class UserController {
                     await redirectOnInvalidSession();
                     return;
                 }
-                await this.handleTosProceedForNewUser(req, res, authFlow, tosFlowInfo, req.body);
-            } else if (TosFlow.WithUser.is(tosFlowInfo)) {
+
+                // there is a possibility, that a competing browser session already created a new user account
+                // for this provider identity, thus we need to check again, in order to avoid created unreachable accounts
+                const user = await this.userService.findUserForLogin({ candidate: tosFlowInfo.candidate });
+                if (user) {
+                    log.info(`(TOS) User was created in a parallel browser session, let's login...`, { logPayload });
+                    await this.loginCompletionHandler.complete(req, res, { user, authHost: tosFlowInfo.authHost, returnToUrl: authFlow.returnTo });
```

Comment:
fyi, this is new: if we retry searching for a user and find one, we definitely should not create new account, but login in into the existing one.